### PR TITLE
Feature/18290 check data service

### PR DESCRIPTION
**Pull request by GuyHarwood**
refactor of check data service

### CODE REVIEW

File: admin/services/data-access/check.data.service.js [hunk 1]
```diff
@@ -46,6 +87,30 @@ checkDataService.findFullyPopulated = async function (criteria) {
...
 
+/**
+ * Find Checks and associated data by checkCode
+ * @param checkCodes - array of UUID
+ * @return {Promise.<void>} - lean Check objects, fully populated
+ * This includes the pupil (includes the school), checkWindow, and checkForm.  This is fairly efficient
+ * involving 1 extra query per document set per sub-document.  The whole lot is done in 5 queries total.
+ */
+checkDataService.sqlFindFullyPopulated = async function (checkCodes) {
+  let sql = 'SELECT * FROM mtc_admin.[check] chk INNER JOIN mtc_admin.pupil pup ON pup.id = chk.pupil_id INNER JOIN mtc_admin.school sch ON sch.id = pup.school_id INNER JOIN mtc_admin.checkWindow wdw ON wdw.id = chk.checkWindow_id INNER JOIN mtc_admin.checkForm frm ON frm.id = chk.checkForm_id'
```
Comment (jon-shipley) [inline admin/services/data-access/check.data.service.js]: This is a bit long - would you consider using backticks  and breaking it up to be more legible?

File: admin/services/marking.service.js [hunk 2]
```diff
@@ -51,7 +53,7 @@ markingService.mark = async function (completedCheck) {
   await completedCheckDataService.save(completedCheck)
 
   // update the check meta info
-  await checkDataService.update({checkCode: completedCheck.data.pupil.checkCode}, {'$set': {results: results}})
+  await checkDataService.sqlSetResults(completedCheck.data.pupil.checkCode, results.marks, results.maxMarks, results.processedAt)
```
Comment (jon-shipley) [inline admin/services/marking.service.js]: I found this name a little odd.   How about  `sqlUpdateCheckWithResults()` ?

File: admin/services/pin-generation.service.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ pinGenerationService.getPupils = async (schoolId, sortField, sortDirection) => {
  * @returns {Boolean}
  */
 pinGenerationService.isValid = async (p) => {
-  const checkCount = await checkDataService.count({ pupilId: p._id, checkStartedAt: { $ne: null } })
+  const checkCount = await checkDataService.sqlNumberOfChecksStartedByPupil(p._id)
```
Comment (jon-shipley) [inline admin/services/pin-generation.service.js]: I feel there should be a `get` or `fetch` in there somewhere.  Or a `find`... so many to choose from.

File: admin/services/pin.service.js [hunk 4]
```diff
@@ -44,8 +44,9 @@ pinService.getActiveSchool = async (schoolId) => {
 pinService.expirePupilPin = async (token, checkCode) => {
   const decoded = jwtService.decode(token)
   const pupil = await pupilDataService.findOne({_id: ObjectId(decoded.sub)})
+  // TODO should this use date service???
   const currentTimeStamp = moment.utc()
-  await checkDataService.update({checkCode: checkCode}, { '$set': { checkStartedAt: currentTimeStamp } })
+  await checkDataService.sqlSetCheckStartedAt(checkCode, currentTimeStamp)
```
Comment (jon-shipley) [inline admin/services/pin.service.js]: This is consistent with the previous, so if you change the other this is a reminder for this one.

File: admin/services/data-access/check.data.service.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/services/data-access/check.data.service.js]: 👍

File: admin/services/marking.service.js (hunk 2 above)
Comment (GuyHarwood) [reply admin/services/marking.service.js]: 👍

File: admin/services/pin-generation.service.js (hunk 3 above)
Comment (GuyHarwood) [reply admin/services/pin-generation.service.js]: 👍

**Pull request merged**